### DISCUSSION TITLE
How to configure Doris to treat the entire JSON string as a value for a field in Kafka stream load?

### DISCUSSION THREAD

**Discussion by bigcloud2008** [apache/doris#30187] [A - General / Q&A]
比如我的kafka主题中json为{"vehicleId":"A1001", "timestamp":1704038400000, "level":1, "name":"mike", "food":"cake"}
我希望整个json的内容作为表的一列mydata写入，表为
CREATE TABLE shops
(

  `vehicleId`       varchar(20) ,
  `timestamp`       LARGEINT(20),
  runDate           Datetime,
  mydata            String
)
ENGINE=olap
UNIQUE KEY( `vehicleId`, `timestamp`, `runDate` )  
PARTITION BY RANGE( runDate  )   
(
) 
DISTRIBUTED BY HASH(`vehicleId`) BUCKETS 10
下面我配置了stream load任务
CREATE ROUTINE LOAD qdyk.mycarbu ON shops
COLUMNS( vehicleId, timestamp, mydata, runDate=date_trunc(from_unixtime(timestamp/1000, 'yyyy-MM-dd HH:mm:ss'), 'day') ),
PROPERTIES
(
    "desired_concurrent_number"="3",
    "max_batch_interval" = "20",
    "max_batch_rows" = "300000",
    "max_batch_size" = "209715200",
	"max_error_number"="999999999",
    "strict_mode" = "false",
    "format" = "json",
    "jsonpaths" = "[\"$.vehicleId\",\"$.timestamp\",\".\"]",
    "strip_outer_array" = "false"
)
FROM KAFKA
(
    "kafka_broker_list" = "10.100.10.101:9092",
    "kafka_topic" = "rc_shops",
	"property.group.id" = "myshops",
	"property.kafka_default_offsets"="OFFSET_BEGINNING"       
);
其中    "jsonpaths" = "[\"$.vehicleId\",\"$.timestamp\",\".\"]",中我希望.代表整个JSON串内容，但是怎么试验，最后写入的都是null，
请问怎么设置jsonpath才能使得一条消息会成为mydata这列的内容，，我试验了$.*, $*等各种写法都不行，都是null， 谢谢

**Comment by LemonLiTree** [comment]
`COLUMNS(k1, k2, k3, vehicleId=k1, timestamp=k2,  runDate=date_trunc(from_unixtime(k2/1000, 'yyyy-MM-dd HH:mm:ss'), 'day'), mydata=k3),`

**Reply by bigcloud2008** [reply]
@LemonLiTree，  你那种写法，原始json中必须有k1,k2,k3列，实际工程中json格式是可变的，一般会有50个字段不会同时出现，按这种写法得配置50个列，非常麻烦，能否有个特殊的表示方法，比如 $ROOT表示整个json字串作为一列呢？

**Reply by LemonLiTree** [reply]
`"json_root" = "$.RECORDS"`
加这个配置指定试试

**Comment by bigcloud2008** [comment]
@LemonLiTree 试验了    "jsonpaths" = "[\"$.vehicleId\",\"$.timestamp\",\"$.RECORDS\"]",不行，不识别，对应表的列写入为null，能否在doris的后续版本中增加一个类似$.RECORDS作为完整json串作为一列的功能呢？谢谢！

**Reply by heichong** [reply]
@LemonLiTree 这个有解决办法吗，我也需要这种：通过jsonpaths获取整个json文档，试了$.* $.$ 都不可以。

**Comment by xunux** [comment]
刚好我也需要

**Comment by ixzc** [comment]
it's supported now. you need to use the latest version and add this on propeties
`        "jsonpaths" = "[\"$.\"]",
`
